napalm_nxos/nxos.py

```diff
@@ -67,8 +81,47 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
             self.port = optional_args.get('port', 443)
         elif self.transport == 'http':
             self.port = optional_args.get('port', 80)
+        elif self.transport == 'netmiko_nxos':
+            # Netmiko possible arguments
+            netmiko_argument_map = {
+                'port': None,
+                'verbose': False,
+                'timeout': self.timeout,
+                'global_delay_factor': 1,
+                'use_keys': False,
+                'key_file': None,
+                'ssh_strict': False,
+                'system_host_keys': False,
+                'alt_host_keys': False,
+                'alt_key_file': '',
+                'ssh_config_file': None,
+            }
 
-    def open(self):
+            if (LooseVersion(netmiko_version) >= LooseVersion('2.0.0') or
+                    LooseVersion(netmiko_version) >= LooseVersion('1.1.0')):
+                netmiko_argument_map['allow_agent'] = False
```

Comment:
This can go away and we can just peg the Netmiko requirement to be Netmiko >= 1.1.0 (in other words, just add `allow_agent` to the netmiko_argument_map and get rid of the LooseVersion.

That code was just transition for when I added the `allow_agent` argument.